design/open/kv-query-tool.md

```diff
@@ -0,0 +1,45 @@
+# KV Store Querying Tool
+
+## Scope
+
+Our newly introduced KV Store stores the data in the form of [`Partition Key`, `Key`, `Value`] triplets. The `partition key` and `key` fields are strings stored as byte arrays and the `value` is an object (various types) serialized as protobuf and stored as byte array.
+This data is not easily interpretable for human viewer, whereas our former SQL DB stored the data (mostly) as human readable data, making it easier for developers to look into the DB, and debug various situations. It also provided the same ease to insert data or alter existing data in the DB for experimentation and, in some cases, as a fix
+With our new KV Store this became less trivial and this proposal comes to offer a solution for that
+
+## Introducing `lakefs kv`
+
+The `lakefs` server contains all the logic needed to correctly read and update the KV Store. It generates objects into byte arrays with their protobuf encoding, on one hand, and rebuilds the objects from byte arrays with protobuf on the other, and, it holds all the logic to generate the relevant keys (e.g. `partition key` from a repository object, branch `key` from the branch object etc.)
+We will elevate this logic and into a new command that will provide the ability to read data items, scan keys and update values, using a new `lakefs` command - `kv`
+```
+lakefs kv <cmd> <args>
+```
+
+### Get/Scan/Set
+
+* `lakefs kv get <partition> <key>` </br>
```

Comment:
Why `lakefs` and not `lakectl`?

Reply:
`lakefs` already has all the logic to access the KV and to manage it's objects.
I think that `lakectl`, as a client, should stay as thin as possible and I don't it should be exposed to very internal details such as KV interface.
Moreover, if we do decide, now or due to some future need, to add ad-hoc commands, such as `scan repository <repository ID>` (as suggested by @eden-ohana), we can leverage graveler to achieve that. This will make `lakectl` even more exposed to `lkaefs` internals
Implementing the client logic in `lakectl` and sending to `lakefs` as a server, seems rather an overkill with API additions, and I don't think we want to get there.